### DISCUSSION TITLE
List of command

### DISCUSSION THREAD

**Discussion by douglasjava** [hyperledger/fabric-admin-sdk#134] [Q&A] [answered]
Hi 
Is there a list of commands we can run to get information in the documentation?
Example: peer node list

**Comment by bestbeforetoday** [comment]
Questions on Fabric CLI commands might be better directed to the [hyperledger/fabric](https://github.com/hyperledger/fabric) repository rather than this client API repository. However, I think you might be looking for the [Command Reference](https://hyperledger-fabric.readthedocs.io/en/latest/command_ref.html) in the Fabric documentation.

**Comment by douglasjava** [comment]
is there any project in go that encapsulates these commands?

**Reply by bestbeforetoday** [reply]
Some of the capability implemented in the CLI commands are recently implemented in Go by [hyperledger/fabric-admin-sdk](https://github.com/hyperledger/fabric-admin-sdk). Primarily the chaincode lifecycle and channel operations.

**Reply by douglasjava** [reply]
Does this project work as an SDK, is there already an example application using it?

**Reply by bestbeforetoday** [reply, marked as answer]
Yes, fabric-admin-sdk works as an SDK. Within the [API documentation](https://pkg.go.dev/github.com/hyperledger/fabric-admin-sdk) there is a [code example](https://pkg.go.dev/github.com/hyperledger/fabric-admin-sdk@v0.0.0-20230602120236-1714d0b04463/pkg/chaincode#example-package) and description of all the functions to support chaincode lifecycle. Other packages are currently less well documented. The [scenario test](https://github.com/hyperledger/fabric-admin-sdk/blob/main/test/e2e_test.go) provides some kind of example of real usage. All of this information is in the project README.

**Comment by bestbeforetoday** [comment]
Closing this discussion as it all relates to other repositories, and discussion should be addressed in those repositories.

**Comment by bestbeforetoday** [comment]
Transferred from fabric-gateway repository and reopened.

**Reply by SamYuan1990** [reply]
thanks, @bestbeforetoday , as I discussed with @denyeart before, I suppose admin-sdk just plays role as sdk, and there is another repo as fabric-binary?

**Reply by bestbeforetoday** [reply]
There is an existing [hyperledger/fabric-cli](https://github.com/hyperledger/fabric-cli) project, which is built on the legacy Go SDK. It might make sense for CLI development to happen there with it refactored to use fabric-admin-sdk instead of fabric-sdk-go. Ideally it would eventually entirely replace the `peer` CLI, which is implemented within the core [fabric](https://github.com/hyperledger/fabric) repository.

I guess a CLI implementation could be done within this repository too. That might make it easier to keep the CLI in step with the latest development stream of the SDK. It might also mean having more dependencies than necessary in this repository though, with people only wanting to use the SDK having to pull in CLI code and dependencies. @denyeart and other Fabric maintainers likely have a view on this.

**Comment by douglasjava** [comment]
this project example exists in the method above, where file the .pem is found on test network, because in project JAVA 

is 

` private static Path getPrivateKeyPath() throws IOException {
        try (var keyFiles = Files.list(KEY_DIR_PATH)) {
            log.info("Path PrivateKey: {} ", keyFiles.findFirst().orElseThrow());
            return keyFiles.findFirst().orElseThrow();
        }
    }
`


`func readPrivateKey() crypto.PrivateKey {
	privateKeyPEM, err := os.ReadFile("privateKey.pem")
	panicOnError(err)

	block, _ := pem.Decode(privateKeyPEM)
	if block == nil {
		panic("failed to parse private key PEM")
	}

	privateKey, err := x509.ParsePKCS8PrivateKey(block.Bytes)
	if err != nil {
		panic("failed to parse PKCS8 encoded private key: " + err.Error())
	}

	return privateKey
}`

PrivateKey exist 

cd /home/developer/hyperledger/fabric-samples/test-network/organizations/peerOrganizations/org1.example.com/users/User1@org1.example.com/msp/keystore
   /hyperledger/fabric-samples/test-network/organizations/peerOrganizations/org1.example.com/users/User1@org1.example.com/msp/keystore$ ls
1023719fce9687f1cf65ef40aba99807f042239e326c7a7c79da495bc10d836c_sk
developer@mibh0744:~/hyperledger/fabric-samples/test-network/organizations/peerOrganizations/org1.example.com/users/User1@org1.example.com/msp/keystore$ vi 1023719fce9687f1cf65ef40aba99807f042239e326c7a7c79da495bc10d836c_sk
